### ISSUE TITLE
Groups of connected shapes can't be copied without a bug

### ISSUE THREAD

**Issue by Romarain** [evolus/pencil#529]
* Operating system: Windows 10
* Pencil version: 3.0.4

Hi,

I tried to build an "image" shape for wireframing purpose, by connecting two free lines (with no arrows at tail ends) to the corners of a rectangle shape, to draw a cross. It worked as intended : after grouping the whole, the resizing of the final shape makes the lines to follow the corners.

Proud of this, I added the shape to my library, but when I tried to load more of one instance of that shape, a strange bug occured (wich occurs if I simply copy the loaded shape on the board too) : the lines try to connect to the first instance of the previous loaded shape ! This making the sub-group of these lines to "outfit" the whole group to wich they belong.

This bug makes the use of the shape impossible, or limited to one instance only, what is useless. I think that it occurs because the lines "memorized" their connections to corners in relation with the rectangle ID ; and when loading a new instance or copying the shape, the rectangle ID of the second instance is the same than the first one.

In short, to fix that problem, you should find a way to generate random IDs of shapes and "update" connection locations. I give a look at the XML of the library and found the shape, but it's too complicated for me to understand and I didn't found any "connection location" related to the master shape ID.
